protocol-spec/generic.md

```diff
@@ -134,5 +142,150 @@
 ]
 ```
 
+## VibrateCmd
+
+**Description:** Causes a toy that supports vibration to run specific vibration motors at a certain speeds. In order to abstract the dynamic range of different toys, the values are sent as a dictionary of vibration motor indexes against floats with a range of \[0.0-1.0\]
+
+**Introduced In Version:** 1
+
+**Message Version:** 1
+
+**Fields:**
+
+* _Id_ \(unsigned int\): Message Id
+* _DeviceIndex_ \(unsigned int\): Index of device
+* _Speeds_ \(array\): Vibration speeds
+  * _Index_ \(unsigned int\): Index of vibration motor
+  * _Speed_ \(double\): Vibration speed with a range of \[0.0-1.0\]
+
+**Expected Response:**
+
+* Ok message with matching Id on successful request.
+* Error message on value or message error.
+
+**Flow Diagram:**
+
+**Serialization Example:**
+
+```json
+[
+  {
+    "VibrateCmd": {
+      "Id": 1,
+      "DeviceIndex": 0,
+      "Speeds": [
+        {
+          "Index": 0,
+          "Speed": 0.5
+        },
+        {
+          "Index": 1,
+          "Speed": 1
+        }
+      ]
+    }
+  }
+]
+```
+
+## LinearCmd
+
+**Description:** Causes a toy that supports linear movement to reposition it's linear actuators at a certain speeds. In order to abstract the dynamic ranges (both speed and movement) of different toys, the values are sent as a dictionary of linear actuator indexes against objects encapsulating floats with a range of \[0.0-1.0\] for speed and position.
```

Comment:
Rewording: 

Causes a toy that supports linear movement to move to a position over a certain amount of time. Devices with multiple linear actuator features may take multiple values. The FeatureCount attribute for the message in the DeviceList/DeviceAdded message will contain that information.